src/backend/language.cpp

```diff
@@ -79,238 +80,1049 @@ LanguageMap::LanguageMap() {
-    //  addLanguage({"cv"}, QT_TRANSLATE_NOOP("QObject", "Chuvash"));
-    addLanguage({"cy"}, QT_TRANSLATE_NOOP("QObject", "Welsh"));
-    addLanguage({"da"}, QT_TRANSLATE_NOOP("QObject", "Danish"));
-    addLanguage({"de"}, QT_TRANSLATE_NOOP("QObject", "German"));
+    addLanguage({"cub"}, QT_TRANSLATE_NOOP("QObject", "Cubeo"));
+    addLanguage({"cuc"}, QT_TRANSLATE_NOOP("QObject", "Usila Chinantec"));
+    addLanguage({"cui"}, QT_TRANSLATE_NOOP("QObject", "Cuiba"));
+    addLanguage({"cuk"}, QT_TRANSLATE_NOOP("QObject", "San Blas Kuna"));
+    addLanguage({"cut"}, QT_TRANSLATE_NOOP("QObject", "Teutila Cuicatec"));
+    addLanguage({"cux"}, QT_TRANSLATE_NOOP("QObject", "Tepeuxila Cuicatec"));
+    addLanguage({"cv-Cyrl"}, QT_TRANSLATE_NOOP("QObject", "Chuvash (Cyrillic script)"));
+    addLanguage({"cwe"}, QT_TRANSLATE_NOOP("QObject", "Kwere"));
+    addLanguage({"cya"}, QT_TRANSLATE_NOOP("QObject", "Nopala Chatino"));
+    addLanguage({"cym", "cy"}, QT_TRANSLATE_NOOP("QObject", "Welsh"));
+
+    addLanguage({"daa"}, QT_TRANSLATE_NOOP("QObject", "Dangaléat"));
+    addLanguage({"dad"}, QT_TRANSLATE_NOOP("QObject", "Marik"));
+    addLanguage({"dah"}, QT_TRANSLATE_NOOP("QObject", "Gwahatike"));
+    addLanguage({"dan", "da"}, QT_TRANSLATE_NOOP("QObject", "Danish"));
+    addLanguage({"dar"}, QT_TRANSLATE_NOOP("QObject", "Dargwa"));
+    addLanguage({"dar-Cyrl"}, QT_TRANSLATE_NOOP("QObject", "Dargwa (Cyrillic script)"));
+    addLanguage({"ded"}, QT_TRANSLATE_NOOP("QObject", "Dedua"));
+    addLanguage({"deu", "de"}, QT_TRANSLATE_NOOP("QObject", "German"));
+    addLanguage({"dgc"}, QT_TRANSLATE_NOOP("QObject", "Casiguran Dumagat Agta"));
+    addLanguage({"dgr"}, QT_TRANSLATE_NOOP("QObject", "Dogrib"));
+    addLanguage({"dgz"}, QT_TRANSLATE_NOOP("QObject", "Daga"));
+    addLanguage({"dhg"}, QT_TRANSLATE_NOOP("QObject", "Dhangu-Dhangu"));
+    addLanguage({"dif"}, QT_TRANSLATE_NOOP("QObject", "Dieri"));
+    addLanguage({"dik"}, QT_TRANSLATE_NOOP("QObject", "Southwestern Dinka"));
+    addLanguage({"dji"}, QT_TRANSLATE_NOOP("QObject", "Djinang"));
+    addLanguage({"djk"}, QT_TRANSLATE_NOOP("QObject", "Eastern Maroon Creole"));
+    addLanguage({"djr"}, QT_TRANSLATE_NOOP("QObject", "Djambarrpuyngu"));
+    addLanguage({"dng"}, QT_TRANSLATE_NOOP("QObject", "Dungan"));
+    addLanguage({"dng-Cyrl"}, QT_TRANSLATE_NOOP("QObject", "Dungan (Cyrillic script)"));
+    addLanguage({"dnj"}, QT_TRANSLATE_NOOP("QObject", "Dan"));
+    addLanguage({"dob"}, QT_TRANSLATE_NOOP("QObject", "Dobu"));
+    addLanguage({"dop"}, QT_TRANSLATE_NOOP("QObject", "Lukpa"));
     addLanguage({"dug"}, QT_TRANSLATE_NOOP("QObject", "Duruma"));
-    //  addLanguage({"dz"}, QT_TRANSLATE_NOOP("QObject", "Dzongkha"));
-    addLanguage({"el", "gre", "ell"}, QT_TRANSLATE_NOOP("QObject", "Greek, Modern (1453-)"));
-    addLanguage({"en"}, QT_TRANSLATE_NOOP("QObject", "English"));
-    addLanguage({"en-US"}, QT_TRANSLATE_NOOP("QObject", "American English"));
```

Comment:
Using an underscore is very likely invalid, because `Language::fromAbbrev()` [contains](https://github.com/bibletime/bibletime/blob/c9a9054ec95f12ab658178930c3a96acd77a9122/src/backend/language.cpp#L333) the following assertion at its beginning:

```C++
BT_ASSERT(!abbrev.contains('_')); // Weak check for certain BCP 47 bugs
```